packages/components/src/core/ComplexFilter/index.tsx

```diff
@@ -69,19 +69,23 @@ const ComplexFilter = <
   const [open, setOpen] = useState(false);
   const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
 
-  const [value, setValue] = useState(getInitialValue());
+  const [value, setValueState] = useState(getInitialValue());
   const [pendingValue, setPendingValue] = useState(getInitialValue());
 
-  useEffect(() => {
-    onChange(value);
-    setPendingValue(value);
-  }, [onChange, value]);
+  const setValue = useCallback(
```

Comment:
hmm it feels like `setValue()` will still get re-created when `onChange` is a new arrow function at reach render. So I'm unsure how this solution is different from the existing one. But I'm probably missing something obvious 😆

Reply:
Ahh great discovery! Yeah it seems like we need to update `ComplexFilter.getInitialValue()` to pass prop.value if available 💡 

Thanks so much for finding the root cause 👏 !!